lfu/lfu.go

```diff
@@ -0,0 +1,127 @@
+// Based on https://github.com/dgrijalva/lfu-go/tree/master
```

Comment:
Also to what extent does this qualify as "based on" - do we have open source legal questions here since this will ultimately get used in the agent which is being distributed?